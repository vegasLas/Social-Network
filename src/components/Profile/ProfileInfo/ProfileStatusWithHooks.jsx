import React, { useState, useEffect } from 'react'
import Profile from '../Profile';

const ProfileStatusWithHooks = (props) => {
    let {status: propsStatus, updateStatus} = props
    let [editMode, setEditMode] = useState(false)
    let [status, setStatus] = useState(propsStatus)

    useEffect(() => {
        setStatus(status);
    }, [propsStatus]);

    const activatedEditMode = () => {
        setEditMode(true)
    }
    const deActivatedEditMode = () => {
        setEditMode(false)
        updateStatus(status);
    }

    const onStatusChange = (e) => {
        setStatus(e.currentTarget.value)
    }

    return (
        <div>
            {!editMode && 
            <div>
                <span onDoubleClick = {activatedEditMode}>
                {status || '-----'}
                </span>
            </div>
            }
            {editMode && 
            <div>
                <input
                onChange = {onStatusChange} 
                autoFocus = {true} 
                onBlur = {deActivatedEditMode}
                value = {status} />
            </div>
            }
        </div>
    )
}

export default ProfileStatusWithHooks