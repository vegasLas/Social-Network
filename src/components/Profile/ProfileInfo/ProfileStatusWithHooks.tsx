import React, { useState, useEffect } from 'react'
type PropsType = {
    status: string
    propsStatus: string
    updateStatus: (status: string) => void
}
const ProfileStatusWithHooks: React.FC<PropsType> = (props) => {
    let {status: propsStatus, updateStatus} = props
    let [editMode, setEditMode] = useState(false)
    let [status, setStatus] = useState(propsStatus)

    useEffect(() => {
        setStatus(status);
    }, [propsStatus]);

    const activatedEditMode = () => {
        setEditMode(true)
    }
    const deActivatedEditMode = () => {
        setEditMode(false)
        updateStatus(status);
    }

    const onStatusChange = (e: any) => {
        setStatus(e.currentTarget.value)
    }

    return (
        <div>
            {!editMode && 
            <div>
                <span onDoubleClick = {activatedEditMode}>
                {status || '-----'}
                </span>
            </div>
            }
            {editMode && 
            <div>
                <input
                onChange = {onStatusChange} 
                autoFocus = {true} 
                onBlur = {deActivatedEditMode}
                value = {status} />
            </div>
            }
        </div>
    )
}

export default ProfileStatusWithHooks