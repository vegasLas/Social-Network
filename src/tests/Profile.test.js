import React from 'react';
import ProfileStatusWithHooks from '../components/Profile/ProfileInfo/ProfileStatusWithHooks';
import ProfileStatus from '../components/Profile/ProfileInfo/ProfileStatusDemo';
import { create } from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow, mount } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() })

describe('ProfileStatus component', () => {
    it("status from props should be auuu", () => {
        const component = create(<ProfileStatusWithHooks status='auuu' />)
        const root = component.root;
        expect(root.props.status).toBe('auuu')
    })
    it("input should be displayed in editMode of span", () => {
        const component = create(<ProfileStatusWithHooks status='auuu' />)
        const root = component.root;
        let span = root.findByType("span")
        span.props.onDoubleClick();
        let input = root.findByType("input")
        expect(input.props.value).toBe('auuu')
    })
    it("input should be displayed in editMode of span", () => {
        const component = create(<ProfileStatusWithHooks status='auuu' />)
        const root = component.root;
        let span = root.findByType("span")
        span.props.onDoubleClick();
        let input = root.findByType("input")
        expect(input.props.value).toBe('auuu')
    })
    it("After creation input shouldn't be displayed", () => {
        const component = create(<ProfileStatusWithHooks status='auuu' />)
        const root = component.root;
        expect(() => {
            let input = root.findByType("input")
        }).toThrow()
    })
    /* it("CallBack should be called", () => {
        const mockCallBack = jest.fn()
        const component = create(<ProfileStatusWithHooks status='auuu' updateStatus={mockCallBack} />)
        const instance = component.getInstance()
        instance.deActivatedEditMode()
        expect(mockCallBack.mock.calls.length).toBe(1)
    }) */
    it('After creation <span> should be displayd', () => {
        let component = create(<ProfileStatusWithHooks status = 'auuu' />)
        let root = component.root;
        let spans = root.findByType('span')
        expect(spans).not.toBeNull()
    })
})