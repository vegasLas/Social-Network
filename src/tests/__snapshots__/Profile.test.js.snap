// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileStatus component should match the snapshot 1`] = `"<div><div><span>-----</span></div></div>"`;
